code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1892,6 +1893,110 @@ GLOBAL_LIST_EMPTY(roundstart_races)
+					var/current_reagent = humi.reagents.get_reagent_amount(/datum/reagent/nitryl)
+					humi.reagents.add_reagent(/datum/reagent/nitryl, max(0, 1 - current_reagent))
+					continue
+				if(gas_amount > 5 && gas_amount < 10)
+					to_chat(humi, span_warning("[pick("Your skin feels itchy.", "You feel like burning.")]"))
+				else
+					to_chat(humi, span_warning("[pick("Your skin feels hot.", "You feel like burning.")]"))
+					humi.adjustFireLoss(gas_amount * 0.5 * delta_time)
+			if(/datum/gas/oxygen)
+				if(gas_amount > 50 && environment.temperature > 400)
+					humi.adjust_fire_stacks(gas_amount * 0.1)
+			if(/datum/gas/plasma)
+				if(isplasmaman(humi))
+					humi.adjustBruteLoss(- gas_amount * 0.5 * delta_time)
+					humi.adjustFireLoss(- gas_amount * 0.15 * delta_time)
+					humi.adjustStaminaLoss(- gas_amount * 0.15 * delta_time)
+					continue
+				if(gas_amount > 5)
+					to_chat(humi, span_warning("[pick("Your skin stings.", "You feel like there is something oily on you.", "You feel something crawling on you!", "Your liver starts moving!", "Your right eye seems sad.", "Your left leg want to be right.")]"))
+					humi.hallucination += gas_amount * 0.5 * delta_time
+					humi.hallucination = clamp(humi.hallucination, 0, 100)
+			if(/datum/gas/pluoxium)
+				if(gas_amount > 10)
+					humi.adjustOxyLoss(- gas_amount * 0.15 * delta_time)
+			if(/datum/gas/proto_nitrate)
+				if(gas_amount > 10)
+					to_chat(humi, span_warning("[pick("Your skin burns!", "You want to remove your skin!")]"))
+					humi.apply_status_effect(/datum/status_effect/gas_rash)
+			if(/datum/gas/tritium)
+				if(gas_amount > 1 && gas_amount < 5)
+					to_chat(humi, span_warning("[pick("Your skin feels tingly.", "You feel tingly all over.")]"))
+				else
+					to_chat(humi, span_warning("[pick("Your skin burns!", "You feel weak!")]"))
+					humi.radiation += gas_amount * 0.15 * delta_time
+					var/current_stamina = humi.getStaminaLoss()
+					if(current_stamina > 25)
+						humi.adjustStaminaLoss(gas_amount * 0.15 * delta_time)
+			if(/datum/gas/water_vapor)
+				if(humi.wear_mask?.clothing_flags & BLOCK_GAS_SMOKE_EFFECT || humi.wear_mask?.clothing_flags & MASKCOVERSEYES || humi.head?.flags_cover & HEADCOVERSEYES)
+					continue
+				if(gas_amount > 5 && gas_amount < 15)
+					to_chat(humi, span_notice("You can't see beyond a few meters from you due to the fog."))
+					humi.overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 1)
+				else if(gas_amount < 30)
+					to_chat(humi, span_warning("You can't see almost right next to you due to the fog."))
+					humi.overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 2)
+				else
+					to_chat(humi, span_warning("The fog stops you from seeing around you!"))
+					humi.become_blind(EYES_COVERED)
+				addtimer(CALLBACK(humi, /mob/living/proc/cure_blind, EYES_COVERED), 1 SECONDS * delta_time)
+				addtimer(CALLBACK(humi, /mob/proc/clear_fullscreen, "tint", 0), 1 SECONDS * delta_time)
```

Comment:
I don't like this use of timers either

Reply:
for this i don't know how to fix it, i think i could just make it a status effect similar to the other one, but i'm not sure

Reply:
made it a status effect